core/templates/domain/feature_gating/feature-status-summary-object.factory.ts

```diff
@@ -0,0 +1,79 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating FeatureStatusSummary domain objects.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+export interface FeatureStatusSummaryBackendDict {
+  [featureName: string]: boolean;
+}
+
+/**
+ * Represents the evaluation summary of all feature flags received from the
```

Comment:
Suggest adding some additional details here explaining how this differentiates from the other platform parameter objects (e.g. that this is used for frontend feature value retrieval and that the other objects are only meant for altering the parameters via the admin page).

Reply:
Done, mentioned that it's meant only for front end feature value retrieval for both `ClientContext` and `FeatureStatusSummary`.